src/sql/workbench/api/node/extHostModelViewDialog.ts

```diff
@@ -23,24 +23,66 @@ class DialogImpl implements sqlops.window.modelviewdialog.Dialog {
 	public customButtons: sqlops.window.modelviewdialog.Button[];
 	public readonly onValidityChanged: vscode.Event<boolean>;
 	private _valid: boolean = true;
+	private _modelView: sqlops.ModelView;
+	public handle: number;
 
-	constructor(private _extHostModelViewDialog: ExtHostModelViewDialog) {
+	constructor(private _extHostModelViewDialog: ExtHostModelViewDialog,
+		private _extHostModelView: ExtHostModelViewShape) {
 		this.okButton = this._extHostModelViewDialog.createButton(nls.localize('dialogOkLabel', 'Done'));
 		this.cancelButton = this._extHostModelViewDialog.createButton(nls.localize('dialogCancelLabel', 'Cancel'));
 		this.onValidityChanged = this._extHostModelViewDialog.getValidityChangedEvent(this);
 		this.onValidityChanged(valid => this._valid = valid);
 	}
 
+	public registerContent(handler: (view: sqlops.ModelView) => void): void {
+		let viewId = 'dialog' + this.handle;
```

Comment:
This is a pretty generic view ID and might cause problems if users happen to register dialogs with the same name. It might be good to make it more unique, but probably not too important (same for tab below)

Reply:
the handle doesn't make it unique 

---
In reply to: [188373815](https://github.com/Microsoft/sqlopsstudio/pull/1415#discussion_r188373815) [](ancestors = 188373815)